clients/src/test/java/org/apache/kafka/common/security/authenticator/SaslAuthenticatorTest.java

```diff
@@ -1682,16 +1671,15 @@ public void testCannotReauthenticateAgainFasterThanOneSecond() throws Exception
              * instead
              */
             time.sleep((long) (CONNECTIONS_MAX_REAUTH_MS_VALUE * 1.1));
-            AssertionError exception = assertThrows(AssertionError.class,
+            AssertionFailedError exception = assertThrows(AssertionFailedError.class,
                 () -> NetworkTestUtils.checkClientConnection(selector, node, 1, 1));
             String expectedResponseTextRegex = "\\w-" + node;
             String receivedResponseTextRegex = ".*" + OAuthBearerLoginModule.OAUTHBEARER_MECHANISM;
-            assertTrue(
-                    "Should have received the SaslHandshakeRequest bytes back since we re-authenticated too quickly, " +
-                    "but instead we got our generated message echoed back, implying re-auth succeeded when it " +
-                    "should not have: " + exception,
-                    exception.getMessage().matches(
-                            ".*<\\[" + expectedResponseTextRegex + "]>.*<\\[" + receivedResponseTextRegex + ".*?]>"));
+            assertTrue(exception.getMessage().matches(
+                ".*<" + expectedResponseTextRegex + ">.*<" + receivedResponseTextRegex + ".*?>"),
```

Comment:
The assertion error message formatting is slightly different in JUnit 5, so I had to remove `[` and `]` from the regex.